common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -674,7 +683,10 @@ public void onComplete(String streamId) throws IOException {
           }
           if (partitionInfo.getCurrentMapIndex() < 0) {
             try {
-              writeAnyDeferredBufs();
+              if (deferredBufs != null && !deferredBufs.isEmpty()) {
+                abortIfNecessary();
```

Comment:
Why does this need to be inside the if block, instead of keeping `writeDeferredBufs` as is and just calling `abortIfNecessary` before `writeDeferredBufs`?

Reply:
I was thinking that if a stream is completing but doesn't need to write any deferred bufs or update the metadata  then we should let it complete without failures.